### PR TITLE
Feat: ModularTable now has the ability to use the useSortBy plugin.

### PR DESCRIPTION
**Pull request by moisesbenzan**
## Done

- Added `useSortBy` plugin capabilities to the `ModularTable` component.
- Added `sortable` prop to the `ModularTable` component.

### CODE REVIEW

File: src/components/ModularTable/ModularTable.tsx [hunk 1]
```diff
@@ -97,9 +105,14 @@ function ModularTable({
                       : "",
                   },
                   { ...getHeaderProps?.(column) },
+                  // Only call this if we want it to be sortable too.
+                  sortable ? column.getSortByToggleProps() : undefined,
                 ])}
               >
                 {column.render("Header")}
+                <span>
+                  {column.isSorted ? (column.isSortedDesc ? " 🔽" : " 🔼") : ""}
```
Comment (bartaz) [inline src/components/ModularTable/ModularTable.tsx:114]: We shouldn't introduce new elements and emoji in the table UI.

Vanilla has it's own built-in styles for sortable tables: https://vanillaframework.io/docs/base/tables#sortable
Technically all it needs it adding `aria-sort` attribute on the `th` element (with one of the values of "none" "ascending" "descending"), so ideally `column.isSorted` should affect `aria-sort` on header, not add content to it.

File: src/components/ModularTable/ModularTable.stories.mdx [hunk 2]
```diff
@@ -191,6 +191,98 @@ columns = {
   </Story>
 </Preview>
 
+### Sortable columns
+
+<Preview>
+  <Story name="Default">
```
Comment (bartaz) [inline src/components/ModularTable/ModularTable.stories.mdx:197]: There can't be two "Default" stories, each needs unique name.

```suggestion
  <Story name="Sortable">
```

File: src/types/react-table-config.d.ts [hunk 3]
```diff
@@ -1,12 +1,53 @@
...
 // https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-table/Readme.md
 
-import "react-table";
+import {
+  UseSortByColumnOptions,
+  UseSortByColumnProps,
+  UseSortByHooks,
+  UseSortByInstanceProps,
+  UseSortByOptions,
+  UseSortByState,
```
[the remark this answers is not in the document]
Comment (moisesbenzan) [reply src/types/react-table-config.d.ts]: Yea, no, I pushed a new commit with the only required one. This was more like leaving everything set up for a future implementation but we won't really do much with them now

[the remark this answers is not in the document]
Comment (bartaz) [reply src/types/react-table-config.d.ts]: Oh cool, that's fine. 
Being future proof is OK as well, I was just a bit overwhelmed by all these new additions there, that didn't seem necessary.

But I also know that this library kind of makes you explicit about this stuff.

Thanks for the update though!

**Comment by bartaz** [comment]
@moisesbenzan Nice work! Would it be possible to add at least some basic test for it? To make sure that the new "sortable" prop works as expected, not only by interactive demo?

**Comment by moisesbenzan** [comment]
> @moisesbenzan Nice work! Would it be possible to add at least some basic test for it? To make sure that the new "sortable" prop works as expected, not only by interactive demo?

I'm not familiar with the testing framework used here :/ If you point me to the right direction I might be able to

**Comment by moisesbenzan** [comment]
Added a basic test for the sortable option

File: src/components/ModularTable/ModularTable.test.tsx [hunk 4]
```diff
@@ -42,6 +42,18 @@ it("renders all rows and columns", async () => {
   });
 });
 
+it("renders all rows and columns when the sortable option is set", async () => {
```
Comment (bartaz) [inline src/components/ModularTable/ModularTable.test.tsx:45]: Thanks, sorry, I haven't noticed your previous comment about giving you some tips.

Would it be possible to replicate a simple test like this one from `MainTable`?
https://github.com/canonical/react-components/blob/main/src/components/MainTable/MainTable.test.tsx#L177-L187

At least to verify "aria-sortable" is added to column headers. We can treat that as a minimal confirmation of the `sortable` feature doing something.

If you don't have time, I can do it myself sometime later this week to finalize and merge your PR.

Comment (moisesbenzan) [reply src/components/ModularTable/ModularTable.test.tsx:60]: Made a test following the basic scheme of that linked test case. Thanks!

**Review by bartaz — APPROVED** [review]
Thanks a lot @moisesbenzan, this looks great!

**Pull request merged**